scipy/optimize/tests/test_hungarian.py

```diff
@@ -41,34 +44,379 @@ def test_linear_sum_assignment():
+    munkres.row_saturated += True
+    munkres.marked += True
+    munkres.row_marked += True
+    munkres.col_marked += True
+
+    assert_array_equal(munkres.marked, np.ones((3, 4), dtype=bool))
+    assert_array_equal(munkres.col_saturated, np.array([True, True, True,
+                                                        True], dtype=bool))
+    assert_array_equal(munkres.row_saturated, np.array([True, True, True],
+                                                       dtype=bool))
+    assert_array_equal(munkres.col_marked, np.array([True, True, True, True],
+                                                    dtype=bool))
+    assert_array_equal(munkres.row_marked, np.array([True, True, True],
+                                                    dtype=bool))
+
+    munkres._remove_covers()
+    assert_array_equal(munkres.marked, np.zeros((3, 4), dtype=bool))
+    assert_array_equal(munkres.col_saturated, np.array([False, False, False,
+                                                        False], dtype=bool))
+    assert_array_equal(munkres.row_saturated, np.array([False, False, False],
+                                                       dtype=bool))
+    assert_array_equal(munkres.col_marked, np.array([False, False, False,
+                                                     False], dtype=bool))
+    assert_array_equal(munkres.row_marked, np.array([False, False, False],
+                                                    dtype=bool))
+
+
+def test_aug_paths_1():
+    """ Test the algorithm that finds a maximum matching from a maximal matching via augmenting
+        paths
+    """
+    # Original biadjacency matrix where zeros represent an edge and
+    # non-zero values represent  non-edges
+    munkres = _hungarian.Munkres(np.array([[0, 0, 0, 0, 0],
+                                           [0, 0, 0, 0, 1],
+                                           [0, 0, 0, 1, 1],
+                                           [0, 0, 1, 1, 1],
+                                           [0, 1, 1, 1, 1]], dtype=float))
+    # A maximal matching that is not maximum
+    munkres.marked = np.array([[1, 0, 0, 0, 0],
+                               [0, 1, 0, 0, 0],
+                               [0, 0, 1, 0, 0],
+                               [0, 0, 0, 0, 0],
+                               [0, 0, 0, 0, 0]], dtype=bool)
+    munkres.row_saturated = np.array([True, True, True, False, False],
+                                     dtype=bool)
+    munkres.col_saturated = np.array([True, True, True, False, False],
+                                     dtype=bool)
+
+    munkres._aug_paths()
+    # The resulting (unique) maximum matching
+    marked = np.array([[0, 0, 0, 0, 1],
+                       [0, 0, 0, 1, 0],
+                       [0, 0, 1, 0, 0],
+                       [0, 1, 0, 0, 0],
+                       [1, 0, 0, 0, 0]], dtype=bool)
+    assert_array_equal(munkres.marked, marked)
+
+
+def test_aug_paths_2():
```

Comment:
some low-level testing of `Munkres` may be appropriate, but where possible please try to test at the level of the public API (so `linear_sum_assignment`)

Reply:
I have tried to refactor the test to use the Public API as much as possible. For the remaining tests, if there is a way to do it without directly using the Munkres class, I am missing it.

Reply:
looks good to me now